File: 0136-single-number/0136-single-number.py
from collections import Counter
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # b = Counter(nums)
        # # print(b)
        # for i in b:
        #     if b[i] == 1:
        #         return i
        

        xor = nums[0]
        for i in range(1,len(nums)):
            xor = xor^nums[i]
        return xor

File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        // HashMap<Integer,Integer> mp = new HashMap<>();
        // for (int i=0;i<nums.length;i++){
        //     mp.put(nums[i],mp.getOrDefault(nums[i],0)+1);
        // }
        // for (Map.Entry<Integer,Integer> a:mp.entrySet()){
        //     if (a.getValue()==1){
        //         return a.getKey();
        //     }
        // }

        int xor = nums[0];
        for (int i=1;i<nums.length;i++){
            xor = xor^nums[i];
        }
        return xor;
    }
}